clients/lsp-eslint.el

```diff
@@ -253,10 +253,30 @@ source.fixAll code action."
         first-argument-exist
       (executable-find (cl-first eslint-server-command)))))
 
-(defun lsp-eslint--confirm-local (_workspace _params)
-  ;; don't bother implementing since it will be dropped in next version of the
-  ;; server
-  t)
+(defvar lsp-eslint--stored-libraries (ht)
+  "Hash table defining if a given path to an ESLint library is allowed to run.
+If the value for a key is t, it will be allowed. If it is any other non-nil
+value, it will not. If it is nil, the user will be prompted to allow or deny it.")
+
+(lsp-defun lsp-eslint--confirm-local (_workspace (&eslint:ConfirmExecutionParams :library-path) callback)
+  (if-let (remembered-answer (gethash library-path lsp-eslint--stored-libraries))
+      (if (eq remembered-answer t)
+          (funcall callback 4)
+        (funcall callback 1))
+    (lsp-ask-question
+     (format "Allow lsp-mode to execute %s?" library-path)
+     '("Always" "Yes" "No" "Never")
+     (lambda (response)
+       (cond ((equal response "Always")
```

Comment:
consider using alist to map String->Number or restructoring the code to something like

```
(let ((a "Yes"))
  (funcall 'identity 
           (cond 
            ((string= "Yes" a) 10))))
```